File: data/src/main/java/com/ahn/data/setting/repository/ThemeRepositoryImpl.kt
package com.ahn.data.setting.repository

import com.ahn.data.setting.local.datasource.ThemePreferenceDataSource
import com.ahn.domain.setting.model.ThemeMode
import com.ahn.domain.setting.repository.ThemeRepository
import javax.inject.Inject

class ThemeRepositoryImpl @Inject constructor(private val dataSource: ThemePreferenceDataSource) : ThemeRepository {
    override fun getThemeMode() = dataSource.getThemeMode()

    override suspend fun saveThemeMode(themeMode: ThemeMode) {
        dataSource.saveThemeMode(themeMode)
    }
}

File: ios/Packages/Data/Sources/Data/Setting/Repository/ThemeRepositoryImpl.swift
import Domain

public struct ThemeRepositoryImpl: ThemeRepository {
    private let dataSource: ThemePreferenceDataSource

    public init(dataSource: ThemePreferenceDataSource) {
        self.dataSource = dataSource
    }

    public func themeMode() -> AsyncStream<ThemeMode> {
        dataSource.themeMode()
    }

    public func saveThemeMode(_ themeMode: ThemeMode) async throws {
        await dataSource.saveThemeMode(themeMode)
    }
}
